### PR TITLE
Tests for DNS synthesize functions

### PR DESCRIPTION
**Pull request by jcoglan**
This is James from @neighbourhoodie, I have been working on the STF milestone M1 to improve test coverage, focussing on `resolved`. We have written a lot of tests for this module and we are splitting the work into a number of PRs, one for each test file we've added. Each new test file usually corresponds to a single source file, and exercises the functions in that source file directly.

This PR includes tests for `resolved-dns-synthesize.c`.

### CODE REVIEW

**Comment by jcoglan** [comment]
I believe I have fixed all the issues with the files in this PR -- asan/ubsan problems, gcc/clang differences, linter checks, etc. and the remaining failing CI tasks are failing for reasons other than the changes here. If that's incorrect and there are other things I still need to change, do let me know what else I need to do.

File: src/resolve/test-dns-synthesize.c [hunk 1]
```diff
@@ -0,0 +1,240 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "resolved-dns-answer.h"
+#include "resolved-dns-question.h"
+#include "resolved-dns-rr.h"
+#include "resolved-dns-synthesize.h"
+#include "resolved-manager.h"
+
+#include "log.h"
+#include "tests.h"
```
Comment (bluca) [inline src/resolve/test-dns-synthesize.c:9]: these are all internal headers, so use the same block above to list them, sorted alphabetically

File: src/resolve/test-dns-synthesize.c [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+ * dns_synthesize_answer()
+ * ================================================================ */
+
+TEST(dns_synthesize_answer_empty) {
+        Manager manager = {};
+        _cleanup_(dns_question_unrefp) DnsQuestion *question = NULL;
+        _cleanup_(dns_resource_key_unrefp) DnsResourceKey *key = NULL;
+        _cleanup_(dns_answer_unrefp) DnsAnswer *answer = NULL;
+
+        question = dns_question_new(1);
```
Comment (bluca) [inline src/resolve/test-dns-synthesize.c:41]: assert here and in other calls below

File: src/resolve/test-dns-synthesize.c [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        ASSERT_NOT_NULL(key);
+
+        ASSERT_OK(dns_question_add(question, key, 0));
+
+        ASSERT_TRUE(dns_synthesize_answer(&manager, question, 0, &answer));
+
+        rr = dns_resource_record_new_full(DNS_CLASS_IN, DNS_TYPE_PTR, "1.0.0.127.in-addr.arpa");
+        ASSERT_NOT_NULL(rr);
+
+        rr->ptr.name = strdup("localhost");
```
Comment (bluca) [inline src/resolve/test-dns-synthesize.c:169]: oom/null check

Comment (bluca) [reply src/resolve/test-dns-synthesize.c:169]: same for other strdup() below

**Review by bluca — APPROVED** [review]

**Comment by jcoglan** [comment]
@bluca `strdup()` calls are now checked.

**Review by bluca — APPROVED** [review]

**Comment by bluca** [comment]
merged manually to solve merge conflict

**Pull request closed without merging**